packages/metals-vscode/src/testExplorer/analyzeTestRun.ts

```diff
@@ -94,8 +94,9 @@ function analyzeTestCases(
 
   if (parent) {
     const tests = new Set(testCases.map((t) => t.id as TestName));
-    const failed = Array.from(testCasesResults.values())
-      .filter((result) => !tests.has(result.testName))
+    const failed = Array.from(testCasesResults.entries())
```

Comment:
Could explain a bit about why these chanegs are needed? Also why do we need to drop the parent in the test case id? This doesn't seem related to the display name

Reply:
Oh, the changes in tests are needed per se. We can go either way: test cases can have the parent's name encoded in the id or not. You suggested that maybe it makes more sense not to do it: https://github.com/scalameta/metals-vscode/pull/1411#discussion_r1284598727, so I changed that. One way or another for the names that come from the dap session we need to do the mapping, since depending on the build server the names might have the parent's id or not. Here the names returned by DAP were used (without mapping), which I believe to be an error anyway.

I'm concerned with the id because previously label (which now corresponds to the display name) was used when a test case was run.